test/server/push_notification_integration.spec.ts

```diff
@@ -47,7 +48,12 @@ describe('Push Notification Integration Tests', () => {
             app.use(express.json());
 
             // Endpoint to receive push notifications
-            app.post('/notify', (req: Request, res: Response) => {
+            app.post('/notify', async (req: Request, res: Response) => {
+                const status = req.body.status.state;
+                // Simulate delay for 'submitted' status to test correct ordering of notifications
+                if (status === 'submitted') {
+                    await new Promise(resolve => setTimeout(resolve, 100))
+                }
```

Comment:
Can this be moved into `/notify/:scenario`? Seems like it's there for cases like this.